backend/spec/controllers/spree/admin/products_controller_spec.rb

```diff
@@ -50,6 +50,78 @@
     end
   end
 
+  # regression test for https://github.com/solidusio/solidus/issues/2791
+  context "creating a product" do
+    before(:all) do
+      create(:shipping_category)
+    end
+
+    it "creates a product" do
+      post :create, params: {
```

Comment:
nice job on checking the flash message, but it would be good to check that the object was created as well, you can wrap the entire post block in an expect and then use .to change...etc.